streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -93,8 +93,8 @@ public boolean isActive() {
     public void initializeIfNeeded() {
         if (state() == State.CREATED) {
             StateManagerUtil.registerStateStores(log, logPrefix, topology, stateMgr, stateDirectory, processorContext);
-
             // initialize the snapshot with the current offsets as we don't need to commit then until they change
```

Comment:
Can you move this comment down a line? We should avoid confusion since we actually don't initialize the (flush) snapshot with the current offsets, just the (commit) snapshot.

To be honest, it's already pretty confusing that we initialize the two snapshots differently. Maybe you could add a quick sentence explaining why for our own future reference

Reply:
We maintain two snapshots in standby because its `commitNeeded` is overridden such that we only commit if the offsets has changed since last time committed, whereas in active we have the flag itself to indicate so.

I can add the above explanation for `offsetSnapshotSinceLastCommit`, would that be clearer?

Reply:
Sounds good. I think the most important thing is to not have the comment that says `initialize the snapshot with the current offsets...` right above the line that initializes the snapshot as empty 🙁

Reply:
Ack